README.md

```diff
@@ -51,29 +86,66 @@ jobs:
 <!-- start configuration-options -->
 ```yaml
 - uses: nrwl/nx-set-shas@v1
+  env:
+    # The github token needed to provide access to APIs.
+    # Use ${{ secrets.GITHUB_TOKEN }} as a value
+    #
+    # Required: true
+    GITHUB_TOKEN: ''
+
   with:
     # The "main" branch of your repository (the base branch which you target with PRs).
     # Common names for this branch include main and master.
     #
     # Default: main
     main-branch-name: ''
 
-    # The glob(7) pattern to be provided to `git describe --match` in order to match against
-    # the latest relevant tag on the specified "main" branch.
-    #
-    # The default pattern aligns with the default behavior of the complementary `nrwl/nx-tag-successful-ci-run` action.
-    #
-    # Default: nx_successful_ci_run*
-    tag-match-pattern: ''
-
     # Applies the derived SHAs for base and head as NX_BASE and NX_HEAD environment variables within the current Job.
     #
     # Default: true
     set-environment-variables-for-job: ''
 
-    # By default, if no matching tags are found on the main branch to determine the SHA, we will log a warning and use HEAD~1. Enable this option to error and exit instead.
+    # By default, if no successful workflow run is found on the main branch to determine the SHA, we will log a warning and use HEAD~1. Enable this option to error and exit instead.
     #
     # Default: false
-    error-on-no-matching-tags: ''
+    error-on-no-successful-workflow: ''
+
+    # The ID of the github action workflow to check for successful run or the name of the file name containing the workflow. 
+    # E.g. 'ci.yml'. If not provided, current workflow id will be used
+    #
+    workflow-id: ''
 ```
-<!-- end configuration-options -->
\ No newline at end of file
+<!-- end configuration-options -->
+
+## Background
+
+When we run `affected` command on [Nx](https://nx.dev/), we can specify 2 git history positions - base and head, and it calculates [which projects in your repository changed
+between those 2 commits](https://nx.dev/latest/angular/tutorial/11-test-affected-projects#step-11-test-affected-projects
+). We can then run a set of tasks (like building or linting) only on those **affected** projects.
+
+This makes it easy to set-up a CI system that scales well with the continous growth of your repository, as you add more and more projects.
+
+
+### Problem
+
+Figuring out what these two git commits are might not be as simple as it seems.
+
+On a CI system that runs on submitted PRs, we determine what commits to include in the **affected** calculation by comparing our `HEAD-commit-of-PR-branch`to the commit in main branch (`master` or `main` usually) from which the PR branch originated. This will ensure our PR in whole is always being tested.
```

Comment:
```suggestion
On a CI system that runs on submitted PRs, we determine what commits to include in the **affected** calculation by comparing our `HEAD-commit-of-PR-branch` to the commit in main branch (`master` or `main` usually) from which the PR branch originated. This will ensure the entirety of our PR is always being tested.
```